src/go/rpk/pkg/cli/cmd/group/seek.go

```diff
@@ -0,0 +1,268 @@
+	cmd.Flags().StringVar(&toFile, "to-file", "", "seek to offsets as specified in the file")
+	cmd.Flags().StringArrayVar(&topics, "topics", nil, "only seek these topics, if any are specified")
+	cmd.Flags().BoolVar(&allowNewTopics, "allow-new-topics", false, "allow seeking to new topics not currently consumed (implied with --to-group or --to-file)")
+
+	return cmd
+}
+
+func parseSeekFile(
+	fs afero.Fs, file string, topics map[string]bool,
+) kadm.Offsets {
+	f, err := fs.Open(file)
+	out.MaybeDie(err, "unable to open %q: %v", file, err)
+
+	o := make(kadm.Offsets)
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		line := s.Text()
+		fields := strings.Split(line, " ")
+		if len(fields) == 1 {
+			fields = strings.Split(line, "\t")
+		}
+		if len(fields) != 3 {
+			out.Die("unable to split line %q by space or tab into three fields", line)
+		}
+
+		topic := fields[0]
+		partition, err := strconv.ParseInt(fields[1], 10, 32)
+		out.MaybeDie(err, "unable to parse partition on line %q: %v", line, err)
+		offset, err := strconv.ParseInt(fields[2], 10, 64)
+		out.MaybeDie(err, "unable to parse offset on line %q: %v", line, err)
+
+		o.Add(kadm.Offset{
+			Topic:       topic,
+			Partition:   int32(partition),
+			At:          offset,
+			LeaderEpoch: -1,
+		})
+	}
+	if err := s.Err(); err != nil {
+		out.Die("file %q scan error: %v", file, err)
+	}
+	return o
+}
+
+func seekFetch(
+	adm *kadm.Client, group string, topics map[string]bool,
+) kadm.Offsets {
+	resps, err := adm.FetchOffsets(context.Background(), group)
+	if err == nil {
+		err = resps.Error()
+	}
+	out.MaybeDie(err, "unable to fetch offsets for %q: %v", group, err)
+	if len(topics) > 0 {
+		resps.DeleteFunc(func(o kadm.OffsetResponse) bool { return !topics[o.Topic] })
+	}
+	return resps.Into()
+}
+
+func seek(
+	fs afero.Fs,
+	adm *kadm.Client,
+	group string,
+	to string,
+	toGroup string,
+	toFile string,
+	topics map[string]bool,
+	allowNewTopics bool,
+) {
+	current := seekFetch(adm, group, topics)
+	var commitTo kadm.Offsets
+	if toFile != "" {
+		commitTo = parseSeekFile(fs, toFile, topics)
+	} else if toGroup != "" {
+		commitTo = seekFetch(adm, toGroup, topics)
+	} else { // --to, we need to list offsets currently used, as well as any extra
+		tps := current.TopicsSet()
+		for topic := range topics {
+			if _, exists := tps[topic]; !exists && !allowNewTopics {
+				out.Die("Cannot commit new topic %q without --allow-new-topics.", topic)
+			}
+			tps[topic] = map[int32]struct{}{} // ensure exists
+		}
+		topics := tps.Topics()
+
+		var listed kadm.ListedOffsets
+		var err error
+		switch to {
+		case "start":
+			listed, err = adm.ListStartOffsets(context.Background(), topics...)
+		case "end":
+			listed, err = adm.ListEndOffsets(context.Background(), topics...)
+		default:
+			var milli int64
+			milli, err = strconv.ParseInt(to, 10, 64)
+			out.MaybeDie(err, "unable to parse millisecond %q: %v", to, err)
+			switch len(to) {
+			case len("1622505600"):
```

Comment:
I think this is to exemplify the kind of values that match this case, but for a moment there it got me wondering if the actual value meant something. What do you think about something like
```suggestion
			case 10: // e.g. "1622505600"

```

Reply:
I'll switch to the numbers here, but it is a little bit riskier because it requires counting this length properly, vs. the `len("digits")` compiling to the same constant.